syntax = "proto3";

package NMessages;

enum ReturnCode {
    SUCESS = 0;
    ERROR = 1;
}

message TBuildOptions {
    optional string name = 1;
    optional string pathtodasbootfile = 2;
}

message TRunOptions{
    optional string name = 1;
}

message TStartOptions{
    optional string name = 1;
    optional string id = 2;
}

message TStopOptions {
    optional string name = 1;
    optional string id = 2;
}

message TPsOptions {
    optional bool show_all = 1;
}

message TRmOptions {
    optional string name = 1;
    optional string id = 2;
}

message TExecOptions {
    optional string name = 1;
    optional string id = 2;
    optional bool detach = 3;
}

message TAttachOptions {
    optional string name = 1;
    optional string id = 2;
    optional bool nostdin = 3;
}

message TOptionsWrapper {
    oneof option {
        TBuildOptions buildOptions = 1;
        TRunOptions runOptions = 2;
        TStartOptions startOptions = 3;
        TStopOptions stopOptions = 4;
        TPsOptions psOptions = 5;
        TRmOptions rmOptions = 6;
        TExecOptions execOptions = 7;
        TAttachOptions attachOptions = 8;
    }
}

message TResult {
    ReturnCode code = 1;
    string text = 2;
}
